cpp/dotenv.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Very simple dotenv implementation in C++,
+ * specifically for the use case of running tests in this repository.
+ *
+ * It tries to read `.env` file in the current directory or a custom file.
+ * If no file is found, no error is thrown, instead the environment variable
+ * is expected to be set in the environment.
+ *
+ * If both the file and the environment variable are set, the environment variable
+ * takes precedence.
+ */
+
+#ifndef DOTENV_H
+#define DOTENV_H
+
+#include <algorithm>
+#include <exception>
+#include <filesystem>
+#include <fstream>
+#include <map>
+#include <stdexcept>
+#include <string>
+
+namespace dotenv {
+
+class Dotenv {
+  private:
+    std::string envFile;
+    std::map<std::string, std::string> envFromFile;
+
+    std::map<std::string, std::string> readEnvFromFile() {
+        std::map<std::string, std::string> env;
+        std::ifstream file(envFile);
+        char charsToRemove[] = {'"', '\''};
+        
+        if (!file) {
+            return env;
+        }
+        
+        for (std::string line; std::getline(file, line);) {
+            const auto pos = line.find('=');
+            if (pos != std::string::npos) {
+                const auto key = line.substr(0, pos);
+                auto value = line.substr(pos + 1);
+
+                removeCharsFromString(value, charsToRemove);
+
+                env[key] = value;
+            }
+        }
+
+        return env;
+    }
+
+    static void removeCharsFromString(std::string& str, char* charsToRemove) {
+        for (unsigned int i = 0; i < strlen(charsToRemove); ++i) {
```

Comment:
`strlen()` scans for the terminating NULL-byte in the string to find its length. But above we define a simple char-array that doesn't fully represent a string and as such also doesn't contain a NULL-byte. We probably get lucky here because the stack likely has a lot of zeros that eventually cause us to terminate.

```suggestion
        for (unsigned int i = 0; i < sizeof(charsToRemove); ++i) {
```